### PR TITLE
Use PR head commit SHA instead of merge SHA to get tree hash

### PR DESCRIPTION
**Pull request by matthyx**
/cc @cjwagner

### CODE REVIEW

**Comment by matthyx** [comment]
This time with the correct branch. Sorry @cjwagner I did that too quickly yesterday.

File: prow/plugins/lgtm/lgtm.go [hunk 1]
```diff
@@ -305,16 +305,14 @@ func handle(wantLGTM bool, config *plugins.Configuration, ownersClient repoowner
...
-					log.WithError(err).Errorf("Failed to get commit.")
-				}
-				treeHash := commit.Commit.Tree.SHA
-				log.Info("Adding comment to store tree-hash: " + treeHash)
-				if err := gc.CreateComment(org, repoName, number, fmt.Sprintf(addLGTMLabelNotification, treeHash)); err != nil {
-					log.WithError(err).Errorf("Failed to add comment.")
-				}
+			commit, err := gc.GetSingleCommit(org, repoName, pr.Head.SHA)
+			if err != nil {
+				log.WithError(err).Errorf("Failed to get commit.")
```
Comment (cjwagner) [inline prow/plugins/lgtm/lgtm.go]:
```suggestion
				log.WithField("sha", pr.Head.SHA).WithError(err).Errorf("Failed to get commit.")
```

File: prow/plugins/lgtm/lgtm.go [hunk 2]
```diff
@@ -305,16 +305,14 @@ func handle(wantLGTM bool, config *plugins.Configuration, ownersClient repoowner
...
+			}
+			treeHash := commit.Commit.Tree.SHA
+			log.Info("Adding comment to store tree-hash: " + treeHash)
```
Comment (cjwagner) [inline prow/plugins/lgtm/lgtm.go]:
```suggestion
			log.WithField("tree", treeHash).Info("Adding comment to store tree-hash.")
```

File: prow/plugins/lgtm/lgtm.go [hunk 3]
```diff
@@ -367,7 +365,7 @@ func handlePullRequest(log *logrus.Entry, gc githubClient, config *plugins.Confi
 				}
 			}
 			// Get the current tree-hash
```
Comment (cjwagner) [inline prow/plugins/lgtm/lgtm.go]: We should check if `lastLgtmTreeHash` is non empty before we use an API token to get the commit. If it is empty we have no chance of matching the tree hash.

File: prow/plugins/lgtm/lgtm.go [hunk 4]
```diff
@@ -367,7 +365,7 @@ func handlePullRequest(log *logrus.Entry, gc githubClient, config *plugins.Confi
...
-			commit, err := gc.GetSingleCommit(org, repo, *pe.PullRequest.MergeSHA)
+			commit, err := gc.GetSingleCommit(org, repo, pe.PullRequest.Head.SHA)
 			if err != nil {
 				log.WithError(err).Errorf("Failed to get commit.")
```
Comment (cjwagner) [inline prow/plugins/lgtm/lgtm.go]:
```suggestion
				log.WithField("sha", pe.PullRequest.Head.SHA).WithError(err).Errorf("Failed to get commit.")
```

**Comment by matthyx** [comment]
@cjwagner ready to check :)

File: prow/plugins/lgtm/lgtm.go [hunk 5]
```diff
@@ -366,16 +364,18 @@ func handlePullRequest(log *logrus.Entry, gc githubClient, config *plugins.Confi
...
-			treeHash := commit.Commit.Tree.SHA
-			if treeHash == lastLgtmTreeHash {
-				// Don't remove the label, PR code hasn't changed
-				log.Infof("Keeping LGTM label as the tree-hash remained the same: %s", treeHash)
-				return nil
+			if lastLgtmTreeHash != "" {
+				// Get the current tree-hash
+				commit, err := gc.GetSingleCommit(org, repo, pe.PullRequest.Head.SHA)
+				if err != nil {
+					log.WithField("sha", pe.PullRequest.Head.SHA).WithError(err).Errorf("Failed to get commit.")
```
Comment (cjwagner) [inline prow/plugins/lgtm/lgtm.go]: `Error` not `Errorf`.
This applies to most all of the `Errorf` calls in this file.

**Comment by matthyx** [comment]
> Please also squash your commits.

Sure @cjwagner, but I need your advice on etiquette: now that we have the tree hash feature, is it better to implement your comments into new commits - allowing you to make incremental reviews, or just `--amend` every time as before?

File: prow/plugins/lgtm/lgtm.go (hunk 5 above)
Comment (matthyx) [reply prow/plugins/lgtm/lgtm.go]: I even wonder why this isn't caught by golint...

**Comment by fejta** [comment]
LGTM

**Comment by cjwagner** [comment]
> Sure @cjwagner, but I need your advice on etiquette: now that we have the tree hash feature, is it better to implement your comments into new commits - allowing you to make incremental reviews, or just --amend every time as before?

I think incremental commits are nice for review (particularly for larger reviews), but we definitely want to squash review commits before merge. The tree hash feature allows the author to squash as a follow up without removing the `lgtm` label so we can have a flow like:
Reviewer:
> /lgtm
> /hold
> Please squash before `/hold cancel`

*author squashes*

PR author:
>     /hold cancel

*merge occurs*

**Pull request merged**